homeassistant/components/emoncms/sensor.py

```diff
@@ -149,6 +156,13 @@ def __init__(
         self._sensorid = sensorid
         self._elem = elem
 
+        if unit_of_measurement == "kWh":
+            self._attr_device_class = DEVICE_CLASS_ENERGY
+            self._attr_state_class = STATE_CLASS_TOTAL_INCREASING
+        elif unit_of_measurement == "W":
+            self._attr_device_class = DEVICE_CLASS_POWER
+            self._attr_state_class = STATE_CLASS_MEASUREMENT
```

Comment:
Emoncms also provides things/sensors like, (daily) averages, right?

Reply:
Ah, I think there are two answers to this, but you're right it's not something I had considered.
1. Things/sensors - Yes. There are other non-power possibilities, such as the emonth units which record temperature/humidity.
2. (daily) averages - Yes and no. EmonCMS itself provides functionality similar to statistics as I understand it, however those are only exposed via the UI. Available APIs are only available for un-aggregated feed values, which is what the current integration uses.

Given this, I think perhaps it would make sense for me to update this to support feeds of temperature/humidity for inclusion in statistics? Is this about what you were thinking, @frenck ?

Reply:
No, I was aiming at the fact that for example: the "Average Daily Energy in kWh" sensor, should not have a `STATE_CLASS_TOTAL_INCREASING`. hence the question.
If they are not present, that is fine for now.

Although, for a future PR, it would be better to explicitly define the entity description in Home Assistant (and now spawn the from API data). For example, if now an average sensor would be added by EmonCMS, things go south. That said, that is nice for a future PR.